rfcs/20200809-keras-saved-model-update.md

```diff
@@ -0,0 +1,175 @@
+variables between layers.
+
+This document introduces a change that may impact some users. Please see the [Breaking Changes](#breaking-changes) section for details.
+
+## Motivation
+
+The integration between core SavedModel serialization and serialization of Keras models is interwoven to an extreme, so that the implementation is hard to understand and replicate. Besides the issue of maintainability, this also raises the question of whether it is possible to write serialization frameworks for high-level TensorFlow libraries utilizing SavedModel without in-depth knowledge about SavedModel and access to internal TensorFlow APIs. Thus this proposal contains the changes needed for the separation of the core SavedModel serialization and Keras model serialization.
+
+### Why separate core vs high-level serialization?
+The core SavedModel saver builds an [`SavedObjectGraph`](https://github.com/tensorflow/tensorflow/blob/r2.3/tensorflow/core/protobuf/saved_object_graph.proto#L24) that stores the metadata and connectivity information of every object connected to the root. When loading, this data is used to recreate every object and reattach the edges between each object. This is sufficient when saving and loading `tf.Modules`, `tf.Variables` and `tf.function`s. However, from the experience of implementing serialization and deserialization for Keras layers, using the metadata and children to load a high-level object isn't so simple.
+
+In the Keras library, there are predefined layers and model networks, which can be serialized with a config object. When loading a layer using the config, both the layer and its weights are created. This introduces some delicate concerns:
+* Layers objects must be loaded in different ways depending on whether there is a config or not (custom vs predefined layer)
+* Layer objects must be loaded first, since loading from the config may cause the creation of other nodes in the ObjectGraph.
+* Functional and Sequential models must be loaded after the layers, to reconnect the model layer network.
+* Any objects *not* created during the above steps should still be loaded. Meaning all objects created by a config must be tracked.
+* Dealing with shared objects that can be created by multiple constructors raises lots of complications.
+
+Thus, introducing a new deserializer API that addresses all these concerns is likely to be complicated and clunky. The core SavedModel API should focus on a single task -- that is saving/loading an object graph with low-level objects. Having a solid base API makes it easier to define serialization for high-level frameworks.
+
+### What should high-level serialization look like?
+High level libraries should utilize the core tf.saved_model.save and tf.saved_model.load functions to save and load the TensorFlow object graph and functions. Anything that is not saved or loaded by the core functions should be saved and loaded separately by the library. 
```

Comment:
> Anything that is not saved or loaded by the core functions should be saved and loaded separately by the library

Just to clarify, does "high level libraries" here refer to `tf.keras.Model.save` or "any 3rd party libraries that may want to save/load `tf.keras.Model` objects"?